packages/plans-grid-next/src/components/comparison-grid/hooks/useVisibleGridPlans.ts

```diff
@@ -0,0 +1,67 @@
+import { PlanSlug, comparePlans, isFreePlan } from '@automattic/calypso-products';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { usePlansGridContext } from '../../../grid-context';
+import { GridPlan } from '../../../types';
+
+/**
+ *When plan changes occur, the visible plan order may need to be updated to reflect the new plans.
+ *we must detect this prop change and transform the current visible plan order into a set of plans
+ *that align with the currently selected plan interval without causing glitches or flashes.
+ *
+ *This hook serves as the solution. The hook accomplishes two main tasks:
+ *- Resolves the correct visible plan order, providing it immediately in the current render cycle to prevent glitches.
+ *- Updates the local state with the most up-to-date plan order as required.
+ * @returns The list of gridPlans in the accurate display configuration.
+ */
+function useVisibleGridPlans(): [ GridPlan[], ( visibleGridPlans: PlanSlug[] ) => void ] {
+	const { gridPlansIndex, gridPlans } = usePlansGridContext();
+	const [ visibleGridPlanState, setVisibleGridPlans ] = useState< GridPlan[] >( gridPlans );
```

Comment:
```suggestion
	const [ visibleGridPlans, setVisibleGridPlans ] = useState< GridPlan[] >( gridPlans );
```

why do we need to suffix with `State`?